forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationService.java

```diff
@@ -281,17 +293,38 @@ public Tpp registerTpp(String cn, String registrationRequestJson,
         return tppStoreService.createTpp(tpp);
     }
 
-    public Tpp updateTpp(Tpp tpp, String token, String registrationRequestJson,
-                         JWTClaimsSet ssaClaims, JSONObject ssaJwsJson,
-                         OIDCRegistrationRequest oidcRegistrationRequest,
-                         String directoryId, Set<SoftwareStatementRole> types) {
-        log.debug("Send the OAuth2 dynamic registration request to the AS");
-        OIDCRegistrationResponse oidcRegistrationResponse = amoidcRegistrationService.updateOIDCClient(token, oidcRegistrationRequest, tpp.getClientId());
-        log.debug("Response from the AS: {}", oidcRegistrationResponse);
+    private String getDirectoryIdFromSsaIssuer(String ssaIssuer) throws DynamicClientRegistrationException {
+        String directoryId;
+        if (ssaIssuer.equals(openBankingDirectoryConfiguration.issuerId)) {
+            directoryId = openBankingDirectoryConfiguration.id;
+        }
+        else if (ssaIssuer.equals(forgeRockDirectoryConfiguration.getIssuerID())){
+            directoryId = forgeRockDirectoryConfiguration.id;
+        } else {
+            String errorString = "Invalid SSA issuer. Issuer " + ssaIssuer + " does not match a trusted issuer";
+            log.debug("registerTpp() {}", errorString);
```

Comment:
Could be better log.error instead of log.debug? because contains the errorMessage.